### PR TITLE
Include nemo logo + expand documentation

### PR DESCRIPTION
**Pull request by matzemathics**

### CODE REVIEW

File: astro.config.mjs [hunk 1]
```diff
@@ -45,8 +45,11 @@ export default defineConfig({
   integrations: [
     starlight({
       title: "Nemo Rule Engine",
+      logo: {
+        dark: "./src/assets/nemo-logo-rusty-bright-nomargin.svg",
+        light: "./src/assets/nemo-logo-rusty-nomargin.svg",
+      },
       customCss: ["./src/tailwind.css"],
-      favicon: "/favicon.svg",
```
Comment (mmarx) [inline astro.config.mjs:49]: Don't drop the favicon here, this will get build by the `vite` plugin

File: astro.config.mjs [hunk 2]
```diff
@@ -45,8 +45,11 @@ export default defineConfig({
   integrations: [
     starlight({
       title: "Nemo Rule Engine",
```
Comment (mmarx) [inline astro.config.mjs:47]: title should probably be `Nemo Graph Rule Engine`, in accordance with the logo?

File: public/favicon.svg
Comment (mmarx) [inline public/favicon.svg:1]: Could also just delete this, since it's not used.

**Review by mmarx — CHANGES_REQUESTED** [review]

File: astro.config.mjs (hunk 1 above)
Comment (matzemathics) [reply astro.config.mjs:49]: weird, I thought I deleted that. must have slipped back in

File: astro.config.mjs (hunk 2 above)
Comment (matzemathics) [reply astro.config.mjs:47]: true, however it looks bad in the default design

**Review by mmarx — APPROVED** [review]

**Pull request merged**